src/equations/2d/linear_scalar_advection.jl

```diff
@@ -48,36 +57,36 @@ end
 
 function initial_conditions_sin_sin(x, t, equation::LinearScalarAdvectionEquation2D)
   # Store translated coordinate for easy use of exact solution
-  x_trans = x - equation.advectionvelocity * t
+  x_trans = x_trans_periodic(x - equation.advectionvelocity * t)
 
   scalar = sin(2 * pi * x_trans[1]) * sin(2 * pi * x_trans[2])
   return @SVector [scalar]
 end
 
 function initial_conditions_constant(x, t, equation::LinearScalarAdvectionEquation2D)
   # Store translated coordinate for easy use of exact solution
-  x_trans = x - equation.advectionvelocity * t
+  x_trans = x_trans_periodic(x - equation.advectionvelocity * t)
```

Comment:
Here we do not need to calculate `x_trans` at all, thus I suggest to remove all these lines.